### PR TITLE
Collections

### PR DESCRIPTION
**Pull request by Tolyaspit**
Please look through the changes and approve it)

### CODE REVIEW

File: src/CustomArraylist.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import java.util.Arrays;
+
+public class CustomArraylist<T> {
+    private Object[] elements; // to store elements
+    private int size = 0;      // number of elements
```
Comment (Alexmansar) [inline src/CustomArraylist.java:5]: 0 is not required. Its by default

File: src/CustomArraylist.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            resize();
+        }
+        elements[size++] = element;
+    }
+
+    @SuppressWarnings("unchecked")
+    public T get(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
```
Comment (Alexmansar) [inline src/CustomArraylist.java:23]: you use it twice - so its a candidate for a new method)

File: src/CustomHashSet.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import java.util.LinkedList;
+
+public class CustomHashSet<T> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private LinkedList<T>[] buckets; // Array of buckets (linked lists)
+    // We'll use separate chaining with linked lists to handle hash collisions
+    private int size = 0;
```
Comment (Alexmansar) [inline src/CustomHashSet.java:8]: same here

File: src/CustomHashSet.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        buckets = new LinkedList[INITIAL_CAPACITY];
+    }
+
+    private int getBucketIndex(T key) {
+        int hash = key.hashCode();
+        return Math.abs(hash) % buckets.length;
+    }
+
+    private void resize() {
+        LinkedList<T>[] oldBuckets = buckets;
```
Comment (Alexmansar) [inline src/CustomHashSet.java:21]: move all private methods after public

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@Tolyaspit Hi)) good job in general, just few comments)) please, fix them))

File: src/CustomArraylist.java (hunk 2 above)
Comment (Tolyaspit) [reply src/CustomArraylist.java:23]: oh , then better to make a new method and implement in inside of them right?

Comment (Alexmansar) [reply src/CustomArraylist.java:23]: what exactly? the idea just make this validation in separate method, and just call it)

Comment (Tolyaspit) [reply src/CustomArraylist.java]: Ok)

**Review by Alexmansar — APPROVED** [review]
@Tolyaspit Could be merged)

**Comment by Alexmansar** [comment]
@Tolyaspit Please, start with the next HW) if you have some question, please, ask)

**Pull request merged**